modules/standard/FFTW.chpl

```diff
@@ -146,15 +191,48 @@ module FFTW {
                                      output, sign, flags);
   }
 
+
   // Real-to-complex and complex-to-real planning routines There are
   // two cases that we treat independently here : in-place and
   // out-of-place transforms
 
-  // Out-of-place routines
-  proc plan_dft_r2c(input : [] real(64), output : [] complex(128), flags :c_uint) : fftw_plan
+  /*
+    Create a plan for a real-to-complex, out-of-place DFT.
+
+    :arg input: The input array, which can be of any rank
+    :type input: [] `real(64)`
+
+    :arg output: The output array, whose size and shape must match the input array's, except for the leading dimension which should be n/2 + 1, where n is the size of the input array's leading dimension.  See the `FFTW documentation <http://www.fftw.org/fftw3_doc/Multi_002dDimensional-DFTs-of-Real-Data.html#Multi_002dDimensional-DFTs-of-Real-Data>`_ for more information.
+    :type output: [] `complex(128)`
+    
+    :arg flags: the bitwise-or of any planning-rigor or algorithm-restriction flags that should be used in creating the plan (e.g., :const:`FFTW_MEASURE` ``|`` :const:`FFTW_PRESERVE_INPUT`)
+    :type flags: c_int
+
+    :returns: The :type:`fftw_plan` representing the resulting plan
+  */
+  proc plan_dft_r2c(input : [?Din] real(64), output : [?Dout] complex(128), 
+                    flags :c_uint) : fftw_plan
   {
     param rank = input.rank: c_int;
 
+    if !noFFTWsizeChecks {
+      var error = false;
+
+      for i in 1..rank-1 do
+        error |= Private_FFTW.checkDimMatch(Din, Dout, i, "plan_dft_r2c()");
+
+      {
+        const inputDim = input.domain.dim(rank).size/2+1;
+        const outputDim = output.domain.dim(rank).size;
+        if (inputDim != outputDim) {
```

Comment:
I think the math is right since (a) you wrote it (b) it passes the tests (c) I've thought hard about it.  But I still struggle with an identity like "inputDim = <the actual inputDim> * N".  I had to rewrite one of them to say, e.g.

const inputDim   = input.domain.dim(rank).size;
const outputDim = output.domain.dim(rank).size;
const outputExp = inputDim / 2 + 1

to convince myself that the math was right.  A silly thing perhaps but I found it easier to digest.
No need to alter at this late date but perhaps food for thought.